### PR TITLE
Implemented IDbCommand

### PR DESCRIPTION
**Pull request by kishoreandukuri**
Code changes done for

1) Reverted the changes of earlier commit for CommandTimeout implementation
2) Added db provider comand property
3) removed IDbCommand parameter in CommandTimeout method in CmnDao.cs class

### CODE REVIEW

File: root/programs/C#/Frameworks/Infrastructure/Public/Db/DamODBC.cs [hunk 1]
```diff
@@ -342,7 +357,7 @@ public override void SetSqlByCommand(string commandText, CommandType commandType
             this._cmd.CommandType = commandType;
 
             // システム共通のCommandTimeout値を設定する。
-            this.SetCommandTimeout(this._cmd); // #x-この行
+            this.SetCommandTimeout(DamIDbCommand); // #x-この行
```
Comment (daisukenishino) [inline root/programs/C#/Frameworks/Infrastructure/Public/Db/DamODBC.cs]: Is this fix is necessary?

File: root/programs/C#/Frameworks/Infrastructure/Public/Db/DamOLEDB.cs [hunk 2]
```diff
@@ -342,7 +358,7 @@ public override void SetSqlByCommand(string commandText, CommandType commandType
             this._cmd.CommandType = commandType;
 
             // システム共通のCommandTimeout値を設定する。
-            this.SetCommandTimeout(this._cmd); // #x-この行
+            this.SetCommandTimeout(DamIDbCommand); // #x-この行
```
Comment (daisukenishino) [inline root/programs/C#/Frameworks/Infrastructure/Public/Db/DamOLEDB.cs]: Is this fix is necessary?

**Comment by daisukenishino** [comment]
There are two patterns usage of SetCommandTimeout method at the SetSqlByCommand method.
- this.SetCommandTimeout (this._cmd); // # x- this line
- this.SetCommandTimeout (DamIDbCommand); // # x- this line

File: root/programs/C#/Frameworks/Infrastructure/Public/Db/DamODBC.cs (hunk 1 above)
Comment (kishoreandukuri) [inline root/programs/C#/Frameworks/Infrastructure/Public/Db/DamODBC.cs]: Dear Nishino-san,

this fix is not necessary.

we can keep  this.SetCommandTimeout(this._cmd);

shall we change it to ' this.SetCommandTimeout(this._cmd);'?

Please let us know.

Thanks,

File: root/programs/C#/Frameworks/Infrastructure/Public/Db/DamOLEDB.cs (hunk 2 above)
Comment (kishoreandukuri) [inline root/programs/C#/Frameworks/Infrastructure/Public/Db/DamOLEDB.cs]: Dear Nishino-san,

this fix is not necessary.

we can keep this.SetCommandTimeout(this._cmd);

shall we change it to ' this.SetCommandTimeout(this._cmd);'?

Please let us know.

Thanks,

**Pull request closed without merging**